src/rgw/rgw_rest_pubsub.cc

```diff
@@ -270,6 +270,15 @@ class RGWPSCreateTopicOp : public RGWOp {
     if (ret < 0) {
       return ret;
     }
+    ret = RGWOp::init_processing(y);
+    if (ret < 0) {
+      return ret;
+    }
+
+    if (s->auth.identity->get_account()) {
+      // account users don't consult the existing owner/policy
```

Comment:
what if the topic already exist?
CreateTopic is similar to override. Don't we need to check topic permissions?

Reply:
i copied the behavior for bucket creation here. there we just call `verify_user_permission()` to check that the s3:CreateBucket action is allowed by identity policy. if the bucket exists, we don't consult its bucket policy for permission, so bucket policy cannot deny s3:CreateBucket permissions

similarly, s3:PutObject only requires permissions from the bucket. if it would overwrite an existing object, we don't consult that object's acls for permission

for these Topic apis, requests will always follow the same-account policy evaluation rules. so we don't *require* permission from the topic policy if an identity policy allows it. the topic policy *can* Deny actions that were otherwise allowed by identity policy, but i don't think that should apply to CreateTopic

aws handles this differently https://docs.aws.amazon.com/sns/latest/api/API_CreateTopic.html:

> This action is idempotent, so if the requester already owns a topic with the specified name, that topic's ARN is returned without creating a new topic.

that said, i think i did mess up DeleteTopic permissions. that should call verify_topic_permission() instead of verify_user_permission()

Reply:
does that make sense? what do you guys think?

Reply:
so in event there is conflict between resource_policy and identity_policy, the `identity_policy` takes the precedence ?

Reply:
@kchheda3 not exactly. a Deny from any policy will reject the request. for same-account access, you just need one Allow from either policy type. for cross-account access, you need an Allow from a resource policy and from identity policy

Reply:
> This action is idempotent, so if the requester already owns a topic with the specified name, that topic's ARN is returned without creating a new topic.

i think this behavior is probably the important difference. bucket re-creation essentially works the same way in that it succeeds, but doesn't let you change anything

since our CreateTopic op does overwrite, it probably makes sense to consult the existing topic policy with verify_topic_permission()

Reply:
> since our CreateTopic op does overwrite, it probably makes sense to consult the existing topic policy with verify_topic_permission()

i pushed a commit to do this